packages/svelte/src/compiler/phases/3-transform/client/visitors/RegularElement.js

```diff
@@ -646,22 +648,24 @@ function build_element_attribute_update_assignment(element, node_id, attribute,
  */
 function is_inlinable_expression(nodes, state) {
 	let has_expression_tag = false;
+	let need_to_escape = false;
 	for (let value of nodes) {
 		if (value.type === 'ExpressionTag') {
 			if (value.expression.type === 'Identifier') {
 				const binding = state.scope
 					.owner(value.expression.name)
 					?.declarations.get(value.expression.name);
-				if (!can_inline_variable(binding)) {
+				if (!binding || !can_inline_variable(binding)) {
 					return false;
 				}
+				need_to_escape = need_to_escape || binding.initial?.type !== 'Literal';
 			} else {
 				return false;
 			}
 			has_expression_tag = true;
 		}
 	}
-	return has_expression_tag;
+	return { need_to_escape };
```

Comment:
Really not a fan of a function returning `false | { need_to_escape: boolean }` — it's a messy and confusing type. If a function name begins with `is_` then it ought to return a boolean; if it needs to return more granular information than that then something like `analyze_expression` would probably be better